utils/generic-components.stanza

```diff
@@ -163,6 +163,43 @@ defn smd-query-properties () -> Tuple<KeyValue<String, ?>> :
 public defn remove-duplicate-keys (hs:Seqable<Seqable<KeyValue<String, ?>>>) -> Tuple<KeyValue<String, ?>> :
   to-tuple $ to-hashtable<String, ?>(cat-all(hs))
 
+; ========================================================
+; ========== Handle min/max value searching ==============
+; ========================================================
+
+defn kvs<?K,?V> (xs:Seqable<KeyValue<?K,?V>>) :
+  for x in xs seq : [key(x), value(x)]
+
+defn replace-params (params:Tuple<KeyValue>) -> Tuple<KeyValue> :
+  ; need to check if resistance is passed a Toleranced() value and if
+  ; so, transform it to min-resistance, max-resistance
+  val replaced-params = to-tuple $
+    generate<KeyValue<String, ?>> :
+      for [myk, myv] in kvs(params) do :
+        match(myv) :
+          (t:Toleranced) :
+            switch(myk) :
+              "resistance" :
+                yield("min-resistance" => min-value(myv))
+                yield("max-resistance" => max-value(myv))
+              "capacitance" :
+                yield("min-capacitance" => min-value(myv))
+                yield("max-capacitance" => max-value(myv))
+              "inductance" :
+                yield("min-inductance" => min-value(myv))
+                yield("max-inductance" => max-value(myv))
+          (_) :
+            yield(myk => myv)          
+  replaced-params  
```

Comment:
```
defn replace-params (params:Tuple<KeyValue>) -> Tuple<KeyValue> :
  ; need to check if resistance is passed a Toleranced() value and if
  ; so, transform it to min-resistance, max-resistance
  to-tuple $
    for [myk, myv] in kvs(params) seq-cat :
      match(myv:Toleranced) :
        [append("min-", myk) => min-value(myv),
         append("max-", myk) => max-value(myv)]
      else :
        [myk => myv]
```

Reply:
There could be other parameters that may be toleranced that I don't want to replace, just the keys of resistance, capacitance, and inductance. I will modify your cool suggestion, though.